src/xdp-utils.c

```diff
@@ -372,10 +372,15 @@ xdp_app_info_rewrite_commandline (XdpAppInfo *app_info,
 
           g_ptr_array_add (args, g_strdup_printf ("--command=%s", quoted_command));
 
-          /* Always quote the app ID to make rewriting the file simpler in case
-           * the app is renamed.
+          /* Always quote the app ID if quote_escape is enabled to make
+           * rewriting the file simpler in case the app is renamed.
            */
-          g_ptr_array_add (args, g_shell_quote (app_info->id));
+          if (quote_escape) {
+            g_ptr_array_add (args, g_shell_quote (app_info->id));
+          } else {
+            g_ptr_array_add (args, g_strdup (app_info->id));
+          }
```

Comment:
The actual code change looks fine to me.

Flatpak uses GNU-style indentation:

```suggestion
          if (quote_escape)
            g_ptr_array_add (args, g_shell_quote (app_info->id));
          else
            g_ptr_array_add (args, g_strdup (app_info->id));
```

or if one of the blocks needed more than one line, it would look like this:

```suggestion
          if (quote_escape)
            {
              g_ptr_array_add (args, g_shell_quote (app_info->id));
            }
          else
            {
              g_ptr_array_add (args, g_strdup (app_info->id));
            }
```